testing/mocks/chain.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mocks
+
+import "github.com/onflow/flow-go/model/flow"
+
+type Chain struct {
+	RootFunc         func() (uint64, error)
+	HeaderFunc       func(height uint64) (*flow.Header, error)
+	CommitFunc       func(height uint64) (flow.StateCommitment, error)
+	EventsFunc       func(height uint64) ([]flow.Event, error)
+	TransactionsFunc func(height uint64) ([]*flow.TransactionBody, error)
+	CollectionsFunc  func(height uint64) ([]*flow.LightCollection, error)
```

Comment:
I decided to put collections in front of transactions everywhere, because the come "before" transactions in the system. Maybe you want to apply the same logic. In which case you might also want to put events behind, because they are "after" transactions (generated by them), and commit before header because commits are the first thing used in our indexer.